core/src/test/scala/org/apache/spark/executor/ExecutorSuite.scala

```diff
@@ -402,6 +403,77 @@ class ExecutorSuite extends SparkFunSuite
     assert(taskMetrics.getMetricValue("JVMHeapMemory") > 0)
   }
 
+  test("SPARK-32175: Plugin initialization should start after heartbeater started") {
+    val tempDir = Utils.createTempDir()
```

Comment:
clean it up at the end of the test? (though I know it will be cleaned by shutdown hook anyway.)

Reply:
I'll try to use `withTempDir`.